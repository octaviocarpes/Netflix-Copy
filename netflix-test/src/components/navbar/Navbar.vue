<template>
  <nav>
    <div class="nav-bar-container">
      <div class="nav-bar-content-left">
        <div class="header-logo"><h1>NIETFLIX</h1></div>
        <div class="links">
          <ul>
            <li id="movies-link"><router-link to="movies">Movies</router-link></li>
            <li id="profile-link"><router-link to="profile">Profile</router-link></li>
            <li id="metrics-link"><router-link to="metrics">Metrics</router-link></li>
          </ul>
        </div>
      </div>
      <div class="nav-bar-content-right">
        <div class="user">
          <img
            :src="require('../../assets/img/profile-icon.jpg')"
            alt="profile icon"
          />
          <p>{{ user.username }}</p>
        </div>
        <p @click="logout">Logout</p>
      </div>
    </div>
  </nav>
</template>

<script>
import store from "@/store";
import Storage from "@/storage";

export default {
  name: "Navbar",

  methods: {
    logout() {
      store.dispatch("LOGOUT");
      Storage.endUserSession();
      this.$router.push("/");
    }
  },

  computed: {
    user() {
      return store.state.auth.user;
    }
  }
};
</script>

<style lang="scss" scoped>
@import "./style.scss";
</style>
