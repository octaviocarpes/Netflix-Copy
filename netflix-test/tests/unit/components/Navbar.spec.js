import Vue from "vue";
import Navbar from "@/components/navbar/Navbar.vue";

describe("Navbar.spec.js", () => {
  let cmp, vm;

  beforeEach(() => {
    cmp = Vue.extend(Navbar);
    vm = new cmp({}).$mount();
  });

  it("has the expected html structure", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});