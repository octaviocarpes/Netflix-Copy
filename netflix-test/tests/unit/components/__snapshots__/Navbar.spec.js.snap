// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar.spec.js has the expected html structure 1`] = `
<nav>
  <div
    class="nav-bar-container"
  >
    <div
      class="nav-bar-content-left"
    >
      <div
        class="header-logo"
      >
        <h1>
          NIETFLIX
        </h1>
      </div>
       
      <div
        class="links"
      >
        <ul>
          <li>
            <router-link
              to="movies"
            >
              Movies
            </router-link>
          </li>
           
          <li>
            <router-link
              to="profile"
            >
              Profile
            </router-link>
          </li>
           
          <li>
            <router-link
              to="metrics"
            >
              Metrics
            </router-link>
          </li>
        </ul>
      </div>
    </div>
     
    <div
      class="nav-bar-content-right"
    >
      <div
        class="user"
      >
        <img
          alt="profile icon"
          src=""
        />
         
        <p>
          
        </p>
      </div>
       
      <p>
        Logout
      </p>
    </div>
  </div>
</nav>
`;
